tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,1461 @@
+    this.set( 'currentYear', currentYear );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            content=content
+            currentYear=currentYear
+            viewMode="years"
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '>:first-child .datepicker-switch' ).text().trim(),
+        '2020-2029',
+        'The current decade is set correctly'
+    );
+
+    this.$( '>:first-child .active' ).click();
+
+    assert.equal(
+        this.$( '>:first-child .datepicker-switch' ).text().trim(),
+        currentYear,
+        'The current year is set correctly'
+    );
+});
+
+test( 'Navigating Forward by Month Crosses to Next Year', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 12;
+    this.set( 'currentMonth', currentMonth );
+
+    this.render( hbs`
+        {{sl-calendar
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '>:first-child .datepicker-switch' ).text().trim(),
+        'December ' + currentYear,
+        'The current month is set correctly'
+    );
+
+    this.$( '>:first-child .next' ).click();
+
+    assert.equal(
+        this.$( '>:first-child .datepicker-switch' ).text().trim(),
+        'January ' + ( currentYear + 1 ),
+        'The next month is in the next year'
+    );
+});
+
+test( 'Navigating Backward by Month Crosses to Previous Year', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 1;
+    this.set( 'currentMonth', currentMonth );
+
+    this.render( hbs`
+        {{sl-calendar
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '>:first-child .datepicker-switch' ).text().trim(),
+        'January ' + currentYear,
+        'The current month is set correctly'
+    );
+
+    this.$( '>:first-child .prev' ).click();
+
+    assert.equal(
+        this.$( '>:first-child .datepicker-switch' ).text().trim(),
+        'December ' + ( currentYear - 1 ),
+        'The previous month is in the previous year'
+    );
+});
+
+test( 'All Twelve Months are Displayed in Order', function( assert ) {
+
+    this.render( hbs`
+        {{sl-calendar
+            viewMode="months"
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '>:first-child .month' ).text().trim(),
+        'JanFebMarAprMayJunJulAugSepOctNovDec',
+        'Twelve months are listed in order'
+    );
+});
+
+test( 'Twelve Years are Displayed in Order', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    this.render( hbs`
+        {{sl-calendar
+            viewMode="years"
+            currentYear=currentYear
+        }}
+    ` );
+
+    let yearsSelection = this.$( '>:first-child' ).find( 'td' ).text().trim();
```

Comment:
Why not use .find('span').text().trim() ?  Then won't need the string replace with it.